File: Algorithm/BOJ/C++/Dynamic Programming/9095.cpp
#include <iostream>
#include <vector>

using namespace std;


int solution(int n) {
	vector<int> answer = { 0, 1, 2, 4 };

	for (int i = 4; i < n + 1; i++) {
		answer.push_back((answer[i - 3] + answer[i - 2] + answer[i - 1]));
	}

	return answer[n];
}


int main() {
	int n, input;

	cin >> n;
	for (int i = 0; i < n; i++) {
		cin >> input;
		cout << solution(input) << endl;
	}

}

File: Algorithm/BOJ/Python/Dynamic Programming/9095.py
def solution(n):
    answer = [0, 1, 2, 4]
    
    for idx in range(4, n + 1):
        answer.append((answer[idx -3] + answer[idx -2] + answer[idx -1]))
    
    return print(answer[n])


n = int(input())
for _ in range(n):
    n = int(input())
    solution(n)
